packages/astro/src/core/render-context.ts

```diff
@@ -226,23 +258,18 @@ export class RenderContext {
 	#i18nData?: Pick<APIContext, 'currentLocale' | 'preferredLocale' | 'preferredLocaleList'>;
 
 	get i18nData() {
-		if (this.#i18nData) return this.#i18nData;
-		const {
-			pipeline: { i18n },
-			request,
-			routeData,
-		} = this;
-		if (!i18n)
-			return {
-				currentLocale: undefined,
-				preferredLocale: undefined,
-				preferredLocaleList: undefined,
-			};
-		const { defaultLocale, locales, routing } = i18n;
+		if (this.#i18nData) return this.#i18nData
+		const { pipeline: { i18n }, request, routeData, url } = this;
+		if (!i18n) return {
+			currentLocale: undefined,
+			preferredLocale: undefined,
+			preferredLocaleList: undefined
+		}
+		const { defaultLocale, locales, routing } = i18n
 		return (this.#i18nData = {
-			currentLocale: computeCurrentLocale(routeData.route, locales, routing, defaultLocale),
+			currentLocale: computeCurrentLocale(url.pathname, locales, routing, defaultLocale) ?? computeCurrentLocale(routeData.route, locales, routing, defaultLocale),
```

Comment:
This change will break people's code. `url.pathname` in SSG with contain the path of the file, e.g. `/dist/en/blog.html` that's not what we want. We want the actual "remote" path e.g. `/en/blog`

Also, we don't even want the "file system" path, because it can contain things like `[...dynamic].astro`, and we can't compute the locale anymore.

Hence why I passed the actual `Route` type. We should revert it.